switch/traceext.ml

```diff
@@ -16,23 +16,157 @@
 
 open Lwt
 
-let size = 128
+type message_id = string * int64
 
-let buffer : (int64 * Message_switch_core.Protocol.Event.t) option array =
-  Array.make size None
+module CompactMessage : sig
+  type t
+
+  val of_message : Message_switch_core.Protocol.Message.t -> t
+
+  val to_message : t -> Message_switch_core.Protocol.Message.t
+
+  val truncate_at : int ref
```

Comment:
My personal preference would be to make this module 'purer' and have this ref passed inside a functor parameter